stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -308,24 +310,32 @@ internal class EventHandlerImpl(
                     batch.addChannel(event.channel)
                 }
 
-                is MessageReadEvent -> {
-                    // get the channel, update reads, write the channel
-                    batch.getCurrentChannel(event.cid)?.let {
-                        val updatedChannel = it.apply {
+                // we use syncState to store the last markAllRead date for a given
+                // user since it makes more sense to write to the database once instead of N times.
+                is MarkAllReadEvent -> domainImpl.repos.syncState.apply {
```

Comment:
`ChatDomain::totalUnreadCount` and `ChatDomain::channelUnreadCount` needs to be updated